CMakeLists.txt

```diff
@@ -407,25 +407,44 @@ if (Hydrogen_ENABLE_CUDA)
 endif (Hydrogen_ENABLE_CUDA)
 
 if (Hydrogen_ENABLE_ROCM)
-  set(CMAKE_MODULE_PATH "/opt/rocm/hip/cmake" ${CMAKE_MODULE_PATH})
+  # Prefer explicit cache variables, then env variables.
+  if (ROCM_PATH)
+    set(_INTERNAL_ROCM_PREFIX ${ROCM_PATH})
+  elseif (DEFINED ENV{ROCM_PATH})
+    set(_INTERNAL_ROCM_PREFIX $ENV{ROCM_PATH})
+  else ()
+    # I don't have a better idea here.
```

Comment:
A hacky solution could be to use the path to something like `rocm-smi` to detect the install directory:
```
execute_process(COMMAND bash "-c" "which rocm-smi | sed 's@bin/rocm-smi@@;'" OUTPUT_VARIABLE _INTERNAL_ROCM_PREFIX)
string(STRIP ${_INTERNAL_ROCM_PREFIX} _INTERNAL_ROCM_PREFIX)
```